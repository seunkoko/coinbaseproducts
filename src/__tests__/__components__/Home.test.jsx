import React from 'react';

// third-party libraries
import { BrowserRouter as Router } from 'react-router-dom';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import toJson from 'enzyme-to-json';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import { spy } from 'sinon';

// components
import ConnectedHome, { Home } from '../../Containers/Home';

// actions
// import { getProducts} from '../../actions/productActions';

const resolvePromise = () => Promise.resolve();

describe('Home', () => {
  const mockStore = configureMockStore([thunk]);
  let store = mockStore({
    products: {
      data: [],
      error: {},
      pending: false,
    },
    stats: {
      data: {},
      error: {},
      pending: false,
    },
  });
  const products = [
    {
        "id": "BCH-USD",
        "base_currency": "BCH",
        "quote_currency": "USD",
        "base_min_size": "0.01",
        "base_max_size": "350",
        "quote_increment": "0.01",
        "display_name": "BCH/USD",
        "status": "online",
        "min_market_funds": "10",
        "max_market_funds": "1000000",
    },
    {
        "id": "BCH-BTC",
        "base_currency": "BCH",
        "quote_currency": "BTC",
        "base_min_size": "0.01",
        "base_max_size": "200",
        "quote_increment": "0.00001",
        "display_name": "BCH/BTC",
        "status": "online",
        "min_market_funds": "0.001",
        "max_market_funds": "30",
    },
    {
        "id": "BCH-GBP",
        "base_currency": "BCH",
        "quote_currency": "GBP",
        "base_min_size": "0.01",
        "base_max_size": "120",
        "quote_increment": "0.01",
        "display_name": "BCH/GBP",
        "status": "online",
        "min_market_funds": "10",
        "max_market_funds": "1000000",
    },
  ];
  let props = {
    products: products,
  };

  it('renders correctly when products are supplied', () => {
    const wrapper = mount(
      <Router>
        <Home
          {...props}
          getProducts={resolvePromise}
        />
      </Router>
    );
    const rendering = toJson(wrapper);
    expect(rendering).toMatchSnapshot();
  });

  // it('renders correctly when getProducts Action is supplied', () => {
  //   const wrapper = mount(
  //     <Router>
  //       <Home
  //         {...props}
  //         getProducts={getProducts}
  //       />
  //     </Router>
  //   );
  //   const rendering = toJson(wrapper);
  //   expect(rendering).toMatchSnapshot();
  // });

  it('renders correctly when connected to the store', () => {
    expect(
      mount(
        <Provider store={store}>
          <ConnectedHome
            {...props}
            getProducts={resolvePromise}
          />
        </Provider>
      ).exists()
    ).toBe(true);
  });

  it('renders correctly when connected to the store and products are changed', () => {
    store = {
      ...store,
      products: {
        data: [ ...products ],
        error: {},
        pending: false,
      },
    };

    expect(
      mount(
        <Provider store={store}>
          <ConnectedHome
            {...props}
            getProducts={resolvePromise}
          />
        </Provider>
      ).exists()
    ).toBe(true);
  });

  it('renders correctly when connected to the store and props is not empty', () => {
    const props = {
      products: {
        data: [ ...products ],
        error: {},
        pending: false,
      },
    }

    expect(
      mount(
        <Provider store={store}>
          <ConnectedHome
            {...props}
            getProducts={resolvePromise}
          />
        </Provider>
      ).exists()
    ).toBe(true);
  });

  it('should call handlePageClick method', () => {
    const wrapper = mount(
      <Home
        products={products}
        getProducts={resolvePromise}
      />
    );

    const data = { selected: 1 };
    const handlePageClickSpy = spy(wrapper.instance(), 'handlePageClick');
    wrapper.instance().handlePageClick(data);
    expect(handlePageClickSpy.called).toEqual(true);
  });
});
