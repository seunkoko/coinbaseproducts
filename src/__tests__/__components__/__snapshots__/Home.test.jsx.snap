// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly when products are supplied 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Home
      getProducts={[Function]}
      products={
        Array [
          Object {
            "base_currency": "BCH",
            "base_max_size": "350",
            "base_min_size": "0.01",
            "display_name": "BCH/USD",
            "id": "BCH-USD",
            "max_market_funds": "1000000",
            "min_market_funds": "10",
            "quote_currency": "USD",
            "quote_increment": "0.01",
            "status": "online",
          },
          Object {
            "base_currency": "BCH",
            "base_max_size": "200",
            "base_min_size": "0.01",
            "display_name": "BCH/BTC",
            "id": "BCH-BTC",
            "max_market_funds": "30",
            "min_market_funds": "0.001",
            "quote_currency": "BTC",
            "quote_increment": "0.00001",
            "status": "online",
          },
          Object {
            "base_currency": "BCH",
            "base_max_size": "120",
            "base_min_size": "0.01",
            "display_name": "BCH/GBP",
            "id": "BCH-GBP",
            "max_market_funds": "1000000",
            "min_market_funds": "10",
            "quote_currency": "GBP",
            "quote_increment": "0.01",
            "status": "online",
          },
        ]
      }
    >
      <Header>
        <nav
          className="navbar navbar-expand-lg nav__header"
        >
          <div
            className="contianer header___container"
          >
            <p
              className="donut__header"
            >
              tryDONUT
            </p>
            <p
              className="coinbase__header"
            >
              Coinbase Products
            </p>
          </div>
        </nav>
      </Header>
      <div
        className="container homebody__container"
      >
        <div
          className="home__error"
        >
          No products to display
        </div>
        <nav />
      </div>
    </Home>
  </Router>
</BrowserRouter>
`;
